src/Http/Client/Response.php

```diff
@@ -330,11 +330,13 @@ public function getCookieCollection(): CookieCollection
     public function getCookie(string $name)
     {
         $this->buildCookieCollection();
-        if (!$this->cookies->has($name)) {
```

Comment:
The `has()` and `get()` methods are almost identical with the only different being `has()` return boolean if cookie exists while `get()` returns the cookie itself. So calling `has()` beforehand had no benefit.

Also since `get()` can return `null` pslam about potentially null value for the `$this->cookies->get($name)->getValue()` chain.

Reply:
If there is a has(), a get() should not return null. the whole idea is then to use has() to check for nullable.
I would rather make this more consistent with this pattern, if you dont mind.

Reply:
That can be addressed in a separate issue/PR if needed.

Reply:
https://github.com/cakephp/cakephp/pull/13239